### PR TITLE
🔧  latest librato, audit closing HTTP bodies, add smtp-test command

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

File: cmd/test-smtp/main.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+type Config struct {
+	URL     string `help:"the SMTP formatted URL to use to test sending"`
+	To      string `help:"the email address to send to"`
+	Subject string `help:"the email subject to send"`
+	Body    string `help:"the email body to send"`
+}
+
+func main() {
```
Comment (nicpottier) [inline cmd/test-smtp/main.go:19]: Peeps were complaining about emails not sending so built this to validate their username / passwords were indeed wrong. Could be useful in the future.

File: go.sum [hunk 2]
```diff
@@ -70,12 +70,10 @@ github.com/nyaruka/ezconf v0.2.1 h1:TDXWoqjqYya1uhou1mAJZg7rgFYL98EB0Tb3+BWtUh0=
 github.com/nyaruka/ezconf v0.2.1/go.mod h1:ey182kYkw2MIi4XiWe1FR/mzI33WCmTWuceDYYxgnQw=
 github.com/nyaruka/gocommon v1.1.1 h1:RnQ+kMzN1lA+W0NpkBDd0mGU3UqadJygR3SMpITMYTQ=
 github.com/nyaruka/gocommon v1.1.1/go.mod h1:QbdU2J9WBsqBmeZRuwndf2f6O7rD7mkC0bGn5UNnwjI=
-github.com/nyaruka/goflow v0.52.0 h1:QYiXBdKq8zwYQTDhsZ5YZyuHcvZ8S4ar9tUuU2kdJbE=
-github.com/nyaruka/goflow v0.52.0/go.mod h1:wI+5qZRajxxZvfYXqbmWauTWAJ0GOTTG3TZDkixU/E0=
 github.com/nyaruka/goflow v0.52.1 h1:vEK3yxo8l+R4vxcLmfzITcfXOdILO1FAbBehuqDAYm0=
 github.com/nyaruka/goflow v0.52.1/go.mod h1:wI+5qZRajxxZvfYXqbmWauTWAJ0GOTTG3TZDkixU/E0=
-github.com/nyaruka/librato v0.0.0-20180827155909-cacc769357b8 h1:TOvxy0u6LNTWP3gwbdNVCiByXJupr9ATFdzBnBJ2TY8=
-github.com/nyaruka/librato v0.0.0-20180827155909-cacc769357b8/go.mod h1:huVocfMEHkttMHD4hSr/wjWNyTx/YMzwwajVzV2bq+0=
+github.com/nyaruka/librato v1.0.0 h1:Vznj9WCeC1yZXbBYyYp40KnbmXLbEkjKmHesV/v2SR0=
```
Comment (nicpottier) [inline go.sum:75]: Latest librato fixed leaking HTTP connections.

File: goflow/engine.go [hunk 3]
```diff
@@ -24,15 +24,18 @@ func RegisterClassificationServiceFactory(factory engine.ClassificationServiceFa
...
 // Engine returns the global engine instance for use in mailroom
 func Engine() flows.Engine {
+	// customize the default golang transport
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	t.MaxIdleConns = 32
+	t.MaxIdleConnsPerHost = 8
+	t.IdleConnTimeout = 30 * time.Second
+	t.TLSClientConfig = &tls.Config{
+		Renegotiation: tls.RenegotiateOnceAsClient, // support single TLS renegotiation
+	}
```
Comment (nicpottier) [inline goflow/engine.go:34]: New transport creation pattern so we don't lose Dial timeouts.

File: ivr/nexmo/nexmo.go [hunk 4]
```diff
@@ -303,8 +304,10 @@ func (c *client) RequestCall(client *http.Client, number urns.URN, resumeURL str
 	if err != nil {
 		return ivr.NilCallID, errors.Wrapf(err, "error trying to start call")
 	}
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusCreated {
+		io.Copy(ioutil.Discard, resp.Body)
```
Comment (nicpottier) [inline ivr/nexmo/nexmo.go:310]: Two rules to make sure connections are reused:
 1. call close on the body
 2. make sure entire body is read

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**